[potty10] iP

potty10 opened the pull request:
## Let it go
> “The snow glows white on the mountain tonight.” – Elsa ([source](https://dansilvestre.com/productivity-quotes))

Not a footprint to be seen
- A kingdom of isolation
- And it looks like I'm the queen
- The wind :wind_face:  is howling like this swirling storm inside

Couldn't keep it in, heaven knows I tried

1. Don't let them in, don't let them [see](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html).
2. Be the good girl you always have to be
3. ~~Conceal~~, don't feel, don't let them know

Well, now they **KNOW**

Let it go, let it go

- [x] Can't hold it back anymore
- [ ] Let it go, let it go
- [ ] Turn away and slam the door 🚪 

___
I don't care what they're going to `say()`. 

```python
def let_the_storm_rage_on():
   the.cold.never.bothered.me.anyway()
}
```

Commit:
Add Gradle support

potty10 committed:
Add echo feature

potty10 committed:
Implement list and add feature

potty10 committed:
Add mark as done feature

potty10 committed:
Add support for different task types

potty10 committed:
Added text UI test

potty10 committed:
Add exception handling for user input

potty10 committed:
Add delete feature

potty10 committed:
Add save feature

potty10 committed:
Update Task objects to store LocalDateTime

potty10 committed:
Merge branch 'branch-Level-7'

potty10 committed:
Merge branch 'branch-Level-8'

potty10 committed:
Refactor object-oriented code

potty10 committed:
Refactor Ui class

potty10 committed:
Reorganise packages

potty10 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

potty10 committed:
Update run script in build.gradle

potty10 committed:
Add test for Task

potty10 committed:
Update Storage

potty10 committed:
Add find feature

potty10 committed:
Update JavaDoc

potty10 committed:
Format code to follow coding standard

potty10 committed:
Merge branch 'branch-Level-9'

potty10 committed:
Merge branch 'branch-A-CodingStandard'

potty10 committed:
Merge branch 'branch-A-JavaDoc'

potty10 committed:
Refactor code to follow Checkstyle

potty10 committed:
Merge branch 'branch-A-CheckStyle'

potty10 committed:
Add gui

potty10 committed:
Merge branch 'branch-Level-10'

jjiayyingtt commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,111 @@
+        return result;
+    }
+
+    private void assertTokenLength(List<String> tokens, int length) throws DukeWrongNumberOfArgumentsException {
```
I like that you split all these functions up to different methods.

jjiayyingtt commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,95 @@
+     * @param str A string that represents a Task in disk format.
+     * @return The corresponding Task object.
+     */
+    private static Task fromString(String str) {
```
consistent coding style

jjiayyingtt commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,95 @@
+        return newtask;
+    }
+
+    /**
```
all public classes and methods have JavaDoc

AngJunYang commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,111 @@
+
+    /**
+     * Returns a stream of tokens from the user command.
+     * @param userInput String to be parsed.
```
Perhaps add an empty line between description and parameter section?

AngJunYang commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,111 @@
+        result.add(tokens[0]);
+
+        switch(tokens[0]) {
+        case "list":
```
Perhaps add a //Fallthrough comment here? I noticed the same issue in other places too

AngJunYang commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,95 @@
+public class Storage {
+
+    /**
+     * Recreates a Task object from the disk format.
```
Perhaps add an empty line between description and parameter section?

AngJunYang reviewed (commented):
Code looks good, with afew coding style violations, but excellent effort put into documentation, keep up the good work!

nigelteosw reviewed (commented):
Looks good to me!

gmengqi commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,164 @@
+            case "bye":
+                return this.byeCommand();
+            default:
+                throw new DukeUnknownCommandException("Unknown command");
```
Your code looks good and it is very easy to follow! Perhaps you can consider declare the commands (list, unmark, mark, todo...) to constants to follow the coding standard!

gmengqi commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,164 @@
+                return this.deleteCommand(Integer.parseInt(tokens.get(1)) - 1);
+            case "find":
+                return this.findCommand(tokens.get(1));
+            case "bye":
```
Your code looks good and it is very easy to follow! Perhaps you can consider declare the commands (list, unmark, mark, todo...) to constants to follow the coding standard!

potty10 committed:
Add assertion statements

There are no assertion statements.

Adding assertion statements can enforce program invariance.

Let's add assertion statements to command methods in the Duke class that
enforces token length, so that get() will not try to get a non-existent
element.

potty10 committed:
Update Duke class

Tokens are currently converted to other objects in parseCommand().

Moving the conversion operation to the respective methods improves
readability.

potty10 committed:
Improve code quality

There are code quality problems such as dead code and magic numbers.

Refactor code to improve readability.

potty10 committed:
Merge pull request #1 from potty10/branch-A-Assertions

Add assertion statements

potty10 committed:
Merge branch 'master' into branch-A-CodeQuality

potty10 committed:
Merge pull request #2 from potty10/branch-A-CodeQuality

Improve code quality

potty10 committed:
Add feature to delete muitple tasks

Users can only delete one task at a time from the list,
which can be inconvenient.

We implement the feature to delete multiple tasks in one command.

potty10 committed:
Merge branch 'branch-C-MassOps'

potty10 committed:
Update chatbot response

The chatbot has no personality, because its responses are not what people
would normally say. By adding some personality, the user feels more
comfortable.

Let's change the response to sentences that are more human-like.

potty10 committed:
Fix display datetime bug

Tasks with DateTime objects display them as "2023-02-11-02-02", which
is not user friendly.

Update tasks to display DateTime objects in the format
"2023-02-11T13:10:58"

potty10 committed:
Improve gui

GUI has a boring background color. Dialog boxes also does not increase
in height if text overflows.

Update GUI to have alternating background colors, so that GUI looks
nicer. Dialog boxes now resizes if text exceeds minimum height.

potty10 committed:
Add test for Parser

potty10 committed:
Add user guide

potty10 committed:
Refactor packages

Many classes are stored in the root package.

Move classes to other packages for clarity.

potty10 committed:
Fix checkstyle violation

potty10 committed:
Fix coding style violation
